src/GoodsList.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class GoodsList extends React.Component {
+  state = {
+    arr: [...this.props.items],
+  };
+
+  reverse = () => {
+    this.setState(() => ({
+      arr: [...this.props.items].reverse(),
+    }));
+  }
+
+  sort = () => {
+    this.setState(() => ({
+      arr: [...this.props.items].sort((a, b) => (a < b ? -1 : 1)),
+    }));
+  }
+
+  reset = () => {
+    this.setState(() => ({
+      arr: [...this.props.items],
+    }));
+  }
+
+  length = () => {
+    this.setState(() => ({
+      arr: [...this.props.items].sort((a, b) => a.length - b.length),
+    }));
+  }
+
+  selectLength = (items) => {
+    this.setState(() => ({
+      arr: [...this.props.items].filter(item => items >= item.length),
```

Comment:
```suggestion
      arr: this.props.items.filter(item => items >= item.length),
```
filter не меняет исходный массив

Reply:
<img width="246" alt="image" src="https://user-images.githubusercontent.com/38065883/80943897-f3318e00-8df0-11ea-9fb1-f07e04d2157f.png">
Работа этой кнопки не соответствует условию. Что вообще делает проверка items >= item.length?